src/omv/sensor.c

```diff
@@ -237,8 +237,45 @@ void sensor_init0()
     // Set default quality
     JPEG_FB()->quality = ((JPEG_QUALITY_HIGH - JPEG_QUALITY_LOW) / 2) + JPEG_QUALITY_LOW;
 
+#ifdef PORTENTA
+    I2C_HandleTypeDef i2c;
+    /* Configure I2C */
+
+    i2c.Instance             = SCCB_I2C;
+    i2c.Init.AddressingMode  = I2C_ADDRESSINGMODE_7BIT;
+    i2c.Init.Timing          = SCCB_TIMING;
+    i2c.Init.DualAddressMode = I2C_DUALADDRESS_DISABLED;
+    i2c.Init.GeneralCallMode = I2C_GENERALCALL_DISABLED;
+    i2c.Init.NoStretchMode   = I2C_NOSTRETCH_DISABLED;
+    i2c.Init.OwnAddress1     = 0xFE;
+    i2c.Init.OwnAddress2     = 0xFE;
+
+    HAL_I2C_Init(&i2c);
+
+    if (SCCB_TIMING == I2C_TIMING_FAST) {
+        // Enable FAST mode plus.
+        if (SCCB_I2C == I2C1) {
+            #if defined(I2C_FASTMODEPLUS_I2C1)
+            HAL_I2CEx_EnableFastModePlus(I2C_FASTMODEPLUS_I2C1);
+            #endif
+        } else if (SCCB_I2C == I2C2) {
+            #if defined(I2C_FASTMODEPLUS_I2C2)
+            HAL_I2CEx_EnableFastModePlus(I2C_FASTMODEPLUS_I2C2);
+            #endif
+        } else if (SCCB_I2C == I2C3) {
+            #if defined(I2C_FASTMODEPLUS_I2C3)
+            HAL_I2CEx_EnableFastModePlus(I2C_FASTMODEPLUS_I2C3);
+            #endif
+        } else {
+            #if defined(I2C_FASTMODEPLUS_I2C4)
+            HAL_I2CEx_EnableFastModePlus(I2C_FASTMODEPLUS_I2C4);
+            #endif
+        }
+    }
+#endif
```

Comment:
That's the same as calling:
```suggestion
cambus_init(&sensor.i2c, SCCB_I2C, SCCB_TIMING);
#endif
```

Reply:
Also note the I2C handle in the sensor struct will Not be re-initialized, and that handle is declared on the stack it will be gone after init0.

Reply:
In my understanding before init0 there's no data structure initialized that can be passed to campus init. Since I just need to reinitialise i2c peripheral it's fine to set up a temporary struct in stack just to turn on the peripheral.

Reply:
> In my understanding before init0 there's no data structure initialized that can be passed to campus init

No you should be able to use the I2C handle in `sensor_t` at this point, it's statically allocated. On the first run `sensor_init0()` is called from `main()` which will initialize `sensor->i2c`, and then `sensor_init()` is called it will re-initialize sensor->i2c (so on first run `sensor->i2c` initialized twice, but that's okay). On later runs, only `sensor_init0()` is called and will reinitialize sensor->i2c. So unless I'm missing something, this should be enough to fix the issue, can you give it a try ?
```
    // Set fb_enabled
    JPEG_FB()->enabled = fb_enabled; // controlled by the IDE.
    #ifdef PORTENTA
    cambus_init(&sensor.i2c, SCCB_I2C, SCCB_TIMING);
    #endif
}
```